compiler/src/iree/compiler/Dialect/LinalgExt/IR/TilingInterfaceImpl.cpp

```diff
@@ -2175,6 +2184,315 @@ LogicalResult OnlineAttentionOp::getResultTilePosition(
+  AffineMap partialMaxMap = getPartialResultMap(getMaxMap(), opInfo);
+  AffineMap partialSumMap = getPartialResultMap(getSumMap(), opInfo);
+
+  SmallVector<AffineMap> indexingMaps = getIndexingMapsArray();
+  indexingMaps[getNumDpsInputs()] = partialAccMap;
+  indexingMaps[getNumDpsInputs() + 1] = partialMaxMap;
+  indexingMaps[getNumDpsInputs() + 2] = partialSumMap;
+
+  SmallVector<Value> tiledOperands;
+  SmallVector<Operation *> slices;
+
+  auto appendSlice = [&](Value val, AffineMap map,
+                         ArrayRef<OpFoldResult> offsets) -> LogicalResult {
+    Operation *sliceOp = getPermutedSlice(b, loc, val, map, offsets, sizes);
+    if (!sliceOp) {
+      return emitOpError("failed to get slice");
+    }
+    tiledOperands.emplace_back(sliceOp->getResult(0));
+    slices.push_back(sliceOp);
+    return success();
+  };
+
+  if (failed(appendSlice(getQuery(), getQueryMap(), offsets))) {
+    return failure();
+  }
+  if (failed(appendSlice(getKey(), getKeyMap(), offsets))) {
+    return failure();
+  }
+  if (failed(appendSlice(getValue(), getValueMap(), offsets))) {
+    return failure();
+  }
+
+  tiledOperands.emplace_back(getScale());
+
+  if (Value mask = getMask()) {
+    if (failed(appendSlice(mask, *getMaskMap(), offsets))) {
+      return failure();
+    }
+  }
+
+  // For results, we set offset of the iterated reduction dims to 0.
+  SmallVector<OpFoldResult> initOffsets(offsets);
+  for (int dim : opInfo.getK2Dims()) {
+    initOffsets[dim] = b.getIndexAttr(0);
+  }
+
+  if (failed(appendSlice(init[0], partialAccMap, initOffsets))) {
+    return failure();
+  }
+  if (failed(appendSlice(init[1], partialMaxMap, initOffsets))) {
+    return failure();
+  }
+  if (failed(appendSlice(init[2], partialSumMap, initOffsets))) {
+    return failure();
+  }
+
+  // Get the initial values.
+  ValueRange slicedInits = ArrayRef(tiledOperands).take_back(3);
+
+  auto tiledOp = cast<OnlineAttentionOp>(
+      mlir::clone(b, getOperation(), slicedInits.getTypes(), tiledOperands));
+  tiledOp.setIndexingMapsAttr(b.getAffineMapArrayAttr(indexingMaps));
+
+  return TilingResult{
+      {tiledOp}, SmallVector<Value>(tiledOp->getResults()), slices};
+}
+
+template <typename CombinerOp>
+static linalg::ReduceOp reduceOnK2(OnlineAttentionOp attn, AffineMap partialMap,
+                                   AttentionOpDetail &opInfo, OpBuilder &b,
+                                   Location loc, Value partialResult,
+                                   Value init) {
+  // linalg.reduce's iteration space is the result's iteration space (and
+  // not the operations iteration space). To account for this, permute the
+  // reduction dimensions based on the partial result map.
+  SmallVector<int64_t> partialReductionDims;
+  for (auto [resultNum, dimExpr] : llvm::enumerate(partialMap.getResults())) {
+    unsigned dim = cast<AffineDimExpr>(dimExpr).getPosition();
+    if (llvm::find(opInfo.getK2Dims(), dim) != opInfo.getK2Dims().end()) {
```

Comment:
can we use `llvm::set_intersect` here?

Reply:
I don't think set_intersect works for different data types. I guess i could do ArrayRef on dim, but llvm::find should be fine.